tests/nlu/selectors/test_selectors.py

```diff
@@ -66,3 +69,60 @@ def test_train_selector(pipeline, component_builder, tmpdir):
         assert rank.get("name") is not None
         assert rank.get("confidence") is not None
         assert rank.get("full_retrieval_intent") is not None
+
+
+def test_training_label():
+
+    # use data that include some responses
+    td = load_data("data/examples/rasa/demo-rasa.md")
+    td_responses = load_data("data/examples/rasa/demo-rasa-responses.md")
+    td = td.merge(td_responses)
+
+    rs = ResponseSelector(component_config={TRAIN_ON_TEXT: False})
+    rs.preprocess_train_data(td)
+
+    assert rs.responses == td.responses
+    assert sorted(list(rs.index_label_id_mapping.values())) == sorted(
+        list(td.responses.keys())
+    )
+
+    rs = ResponseSelector(component_config={TRAIN_ON_TEXT: True})
+    rs.preprocess_train_data(td)
+
+    assert rs.responses == td.responses
+    assert sorted(list(rs.index_label_id_mapping.values())) == sorted(
+        [r[0].get("text") for r in td.responses.values()]
+    )
+
+
+@pytest.mark.parametrize(
+    "pred_label, train_on_text, full_response_label, full_response_text",
```

Comment:
what is a `full_response_label` was that not previously a `full_response_key` or `intent_response_key` or `combined_intent_response_key`. 

the naming is quite confusing:
- what is the name of `faq/ask_name`?
- what is the name of `faq`?
- what is the name of `ask_name`?

Reply:
So, `faq` is the `retrieval intent` or just `intent`(depending on the context), `ask_name` is the `response_key` and `faq/ask_name` is the `intent_response_key`.
I'll change the names in the parameters here to better reflect this naming.